### PR TITLE
refactor(docgen): simplify init

### PR DESCRIPTION
**Pull request by cristaloleg**
Adding `add(v any)` and `must` that simplifies life.

### CODE REVIEW

File: api/docgen/examples.go [hunk 1]
```diff
@@ -35,198 +35,185 @@ import (
...
+	add(float64(42))
+	add(float64(42))
+	add(true)
+	add([]byte("byte array"))
+	add(time.Duration(time.Second))
+	add(node.Full)
+	add(auth.Permission("admin"))
+	add(byzantine.BadEncoding)
+
+	// TODO: this case requires more debugging, simple to leave it as it was.
```
Comment (cristaloleg) [inline api/docgen/examples.go:82]: Literally what comments says.

File: api/docgen/examples.go [hunk 2]
```diff
@@ -35,198 +35,185 @@ import (
...
-}
-
-func addToExampleValues(v interface{}) {
-	ExampleValues[reflect.TypeOf(v)] = v
+	))
 }
 
-func ExampleValue(t, parent reflect.Type) (interface{}, error) {
-	v, ok := ExampleValues[t]
+func exampleValue(t, parent reflect.Type) (any, error) {
```
Comment (cristaloleg) [inline api/docgen/examples.go:188]: Unexported 'cause it's not accessible outside.

File: api/docgen/examples.go [hunk 3]
```diff
@@ -35,198 +35,185 @@ import (
...
+	exampleResourceMngrStats string
+
+	//go:embed "exampledata/blob.json"
+	exampleBlob string
+
+	//go:embed "exampledata/blobProof.json"
+	exampleBlobProof string
+)
+
+var exampleValues = map[reflect.Type]any{}
```
Comment (cristaloleg) [inline api/docgen/examples.go:58]: Unexported 'cause it's not accessible outside.

**Review by Wondertan — COMMENTED** [review]
Lint failure, otherwise LGTM

**Review by Wondertan — APPROVED** [review]

**Review by renaynay — APPROVED** [review]
utack

**Pull request merged**